lib/eslint/eslint.js

```diff
@@ -300,7 +305,8 @@ function processOptions({
         reportUnusedDisableDirectives,
         resolvePluginsRelativeTo,
         rulePaths,
-        useEslintrc
+        useEslintrc,
+        ruleFilter
```

Comment:
Looks like the `ESLint` class accepts the new option, but then doesn't use it? We should either implement the change in this class as well, or not allow passing the new option. Per @nzakas's https://github.com/eslint/eslint/issues/16450#issuecomment-1310829909, I believe the idea was to not implement this change in the `ESLint` class, so it should not allow this option to be passed to the constructor.